flocker/control/test/test_httpapi.py

```diff
@@ -2171,42 +2234,143 @@ def saved(ignored):
         saving.addCallback(saved)
         return saving
 
-    def test_no_primary(self):
+
+RealTestsUpdatePrimaryDataset, MemoryTestsUpdatePrimaryDataset = (
+    buildIntegrationTests(
+        UpdatePrimaryDatasetTestsMixin, "UpdatePrimaryDataset", _build_app)
+)
+
+
+class UpdateSizeDatasetTestsMixin(APITestsMixin):
+    """
+    Tests for the behaviour of the dataset modification endpoint at
+    ``/configuration/datasets/<dataset_id>`` when supplied with a maximum_size
+    value.
+    """
+    def test_unknown_dataset(self):
+        """
+        NOT_FOUND is returned if the requested dataset_id doesn't exist.
+        The error includes the requested dataset_id.
+        """
+        unknown_dataset_id = unicode(uuid4())
+        return self.assertResult(
+            method=b"POST",
+            path=b"/configuration/datasets/%s" % (
+                unknown_dataset_id.encode('ascii'),),
+            request_body={},
+            expected_code=NOT_FOUND,
+            expected_result={u"description": u'Dataset not found.'}
+        )
```

Comment:
This seems a lot like `UpdatePrimaryDatasetTestsMixin.test_unknown_dataset`.  And there doesn't seem to be anything actually related to size updates in this test.